plugins/module_utils/gitlab.py

```diff
@@ -110,3 +110,14 @@ def gitlab_authentication(module):
             GitLab remove Session API now that private tokens are removed from user API endpoints since version 10.2." % to_native(e))
 
     return gitlab_instance
+
+
+def filter_returned_variables(gitlab_variables):
+    # pop properties we don't know
+    existing_variables = [dict(x.attributes) for x in gitlab_variables]
```

Comment:
Just a note, `x.attributes` is already a dict (https://github.com/python-gitlab/python-gitlab/blob/b6c08725042380d20ef5f09979bc29f2f6c1ab6f/gitlab/base.py#L139-L149) (and I agree the list comprehension looked a bit more concise). But feel free to ignore this comment.

Probably outside of the scope of this PR but I'm starting to think at some point there could be 1 source of truth for the `KNOWN` list.

Reply:
If `x` is a dict, `dict(x)` creates a copy of it. You could also write `x.copy()`.

Reply:
Ah yes I totally missed that part of the discussion, resolving this